fixtures/error-types/src/lib.rs

```diff
@@ -28,12 +28,10 @@ impl From<anyhow::Error> for ErrorInterface {
 }
 
 // must do explicit conversion...
-fn oops() -> Result<(), Arc<ErrorInterface>> {
-    Err(Arc::new(
-        anyhow::Error::msg("oops")
-            .context("because uniffi told me so")
-            .into(),
-    ))
+fn oops() -> Result<(), ErrorInterface> {
```

Comment:
I assume the old code still works, so maybe keep a test with an explicit `Arc<Struct>`? There's also a trait example below that does keep `Arc<>`, but should it also test a non-arc version?

Maybe also update https://mozilla.github.io/uniffi-rs/udl/errors.html to remove the `Arc<>` from the example, and maybe a changelog entry?

Reply:
> I assume the old code still works, so maybe keep a test with an explicit Arc<Struct>?

That makes sense, I'll update the test code.

> There's also a trait example below that does keep Arc<>, but should it also test a non-arc version?

This isn't currently possible because you need the Arc to handle the `dyn Trait` part.